host/testcluster.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package host
+
+import (
+	"github.com/uber-common/bark"
+	wsc "github.com/uber/cadence/.gen/go/cadence/workflowserviceclient"
+	"github.com/uber/cadence/client"
+	"github.com/uber/cadence/common/blobstore/filestore"
+	"github.com/uber/cadence/common/messaging"
+	"github.com/uber/cadence/common/persistence/persistence-tests"
+	"io/ioutil"
+	"os"
+	"sync"
+)
+
+type (
+	// TestCluster is a base struct for integration tests
+	TestCluster struct {
+		persistencetests.TestBase
+		BlobstoreBase
+		sync.Mutex // This lock is to protect the race in ringpop initialization.
+
+		Host   Cadence
+		engine wsc.Interface
+		Logger bark.Logger
+	}
+
+	// TestClusterOptions are options for test cluster
+	TestClusterOptions struct {
+		PersistOptions   *persistencetests.TestBaseOptions
```

Comment:
I think we should get rid of these and completely rely on dynamic config when starting up the host.  We should be able to create multiple configurations by just providing different configs for the server.